heartbeat/storage-mon.in

```diff
@@ -173,77 +205,182 @@ storage-mon_init() {
 	fi
 }
 
-storage-mon_validate() {
-	storage-mon_init
+storage-mon_update_attribute() {
 
-	# Is the state directory writable?
-	state_dir=$(dirname "$OCF_RESKEY_state_file")
-	touch "$state_dir/$$"
-	if [ $? -ne 0 ]; then
-		return $OCF_ERR_CONFIGURED
-	fi
-	rm "$state_dir/$$"
+	while :
+	do
+		"$ATTRDUP" -n ${ATTRNAME} -U "$1" -d "5s"
+		rc=$?
+		if [ $rc -eq 0 ]; then
+			break
+		fi
 
+		ocf_log debug "${1} attribute by attrd_updater failed"
+		if [ "$1" = "red" ]; then
+			# If the attrd_updater fails with the red attribute, return an error to let pacemaker handle the failure immediately.
+			return $OCF_ERR_GENERIC
+		fi
+	done
 	return $OCF_SUCCESS
 }
 
 storage-mon_monitor() {
-	storage-mon_init
+	if [ -z "$OCF_RESKEY_daemonize" ]; then
```

Comment:
Should be `if ! ocf_is_true "$OCF_RESKEY_daemonize"; then`. Update other occurences as well.